core/appeal/service.go

```diff
@@ -1537,7 +1539,45 @@ func getPolicy(a *domain.Appeal, p *domain.Provider, policiesMap map[string]map[
 	if resourceConfig == nil {
 		return nil, fmt.Errorf("%w: couldn't find %q resource type in the provider config", ErrInvalidResourceType, a.Resource.Type)
 	}
-	policyConfig := resourceConfig.Policy
+	policyConfig = *resourceConfig.Policy
+
+	appealMap, err := a.ToMap()
+	if err != nil {
+		return nil, fmt.Errorf("parsing appeal struct to map: %w", err)
+	}
+
+	var dynamicPolicyConfigData string
+	for _, pc := range p.Config.Policies {
+		if pc.When != "" {
+			v, err := evaluator.Expression(pc.When).EvaluateWithVars(map[string]interface{}{
+				"appeal": appealMap,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			isFalsy := reflect.ValueOf(v).IsZero()
+			if isFalsy {
+				continue
+			}
+
+			dynamicPolicyConfigData = pc.Policy
```

Comment:
we should break the loop once we got the first matching `When` criteria and doesn't continue to the rest of the policies.